library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadAdapter.kt

```diff
@@ -0,0 +1,130 @@
+        holder.bind(bodyItems[position])
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TransactionPayloadViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        return when (viewType) {
+            TYPE_HEADERS -> {
+                val view = inflater.inflate(R.layout.chucker_transaction_item_headers, parent, false)
+                TransactionPayloadViewHolder.HeaderViewHolder(view)
+            }
+            TYPE_BODY_LINE -> {
+                val view = inflater.inflate(R.layout.chucker_transaction_item_body_line, parent, false)
+                TransactionPayloadViewHolder.BodyLineViewHolder(view)
+            }
+            else -> {
+                val view = inflater.inflate(R.layout.chucker_transaction_item_image, parent, false)
+                TransactionPayloadViewHolder.ImageViewHolder(view)
+            }
+        }
+    }
+
+    override fun getItemCount() = bodyItems.size
+
+    override fun getItemViewType(position: Int): Int {
+        return when (bodyItems[position]) {
+            is TransactionPayloadItem.HeaderItem -> TYPE_HEADERS
+            is TransactionPayloadItem.BodyLineItem -> TYPE_BODY_LINE
+            is TransactionPayloadItem.ImageItem -> TYPE_IMAGE
+        }
+    }
+
+    internal fun highlightQueryWithColors(newText: String, backgroundColor: Int, foregroundColor: Int) {
+        bodyItems.filterIsInstance<TransactionPayloadItem.BodyLineItem>()
+            .withIndex()
+            .forEach { (index, item) ->
+                if (newText in item.line) {
+                    item.line.clearSpans()
+                    item.line = item.line.toString()
+                        .highlightWithDefinedColors(newText, backgroundColor, foregroundColor)
+                    notifyItemChanged(index + 1)
+                } else {
+                    // Let's clear the spans if we haven't found the query string.
+                    val spans = item.line.getSpans(0, item.line.length - 1, Any::class.java)
+                    if (spans.isNotEmpty()) {
+                        item.line.clearSpans()
+                        notifyItemChanged(index + 1)
+                    }
+                }
+            }
+    }
+
+    internal fun resetHighlight() {
+        bodyItems.filterIsInstance<TransactionPayloadItem.BodyLineItem>()
+            .withIndex()
+            .forEach { (index, item) ->
+                val spans = item.line.getSpans(0, item.line.length - 1, Any::class.java)
+                if (spans.isNotEmpty()) {
+                    item.line.clearSpans()
+                    notifyItemChanged(index + 1)
+                }
+            }
+    }
+
+    companion object {
+        private const val TYPE_HEADERS = 1
+        private const val TYPE_BODY_LINE = 2
+        private const val TYPE_IMAGE = 3
+    }
+}
+
+internal sealed class TransactionPayloadViewHolder(view: View) : RecyclerView.ViewHolder(view) {
```

Comment:
Could you please explain why this class is `sealed`?

Reply:
It is not a concern. I was just curious why it is `sealed`. But now I got it. Thanks.